composer/profiler/torch_profiler.py

```diff
@@ -202,28 +256,60 @@ def handler_fn(prof: torch.profiler.profiler.profile):
-                                                                        run_name=state.run_name,
-                                                                        timestamp=timestamp)
-                trace_remote_file_name = trace_remote_file_name.lstrip('/')
-                logger.upload_file(remote_file_name=trace_remote_file_name,
-                                   file_path=trace_file_name,
-                                   overwrite=self.overwrite)
+            if self.filename is not None:
+                trace_file_name = os.path.join(
+                    folder_name,
+                    format_name_with_dist_and_time(self.filename, run_name=state.run_name, timestamp=timestamp),
+                )
+                trace_file_dirname = os.path.dirname(trace_file_name)
+                if trace_file_dirname:
+                    os.makedirs(trace_file_dirname, exist_ok=True)
+                prof.export_chrome_trace(trace_file_name)
+                state.profiler.record_chrome_json_trace_file(trace_file_name)
+                if self.remote_file_name is not None:
+                    trace_remote_file_name = format_name_with_dist_and_time(self.remote_file_name,
+                                                                            run_name=state.run_name,
+                                                                            timestamp=timestamp)
+                    trace_remote_file_name = trace_remote_file_name.lstrip('/')
+                    logger.upload_file(remote_file_name=trace_remote_file_name,
+                                       file_path=trace_file_name,
+                                       overwrite=self.overwrite)
+
+            log.debug(f'Memory profiler enabled: {self.memory_filename if self.memory_filename else False}')
+            if self.memory_filename is not None:
+                if version.parse(torch.__version__) > version.parse('2.1.0.dev'):  # type: ignore
+                    memory_trace_file_name = os.path.join(
+                        folder_name,
+                        format_name_with_dist_and_time(self.memory_filename,
+                                                       run_name=state.run_name,
+                                                       timestamp=timestamp),
+                    )
+                    log.debug(f'Saving memory trace to {memory_trace_file_name}')
+                    memory_trace_file_dirname = os.path.dirname(memory_trace_file_name)
+                    if memory_trace_file_dirname:
+                        os.makedirs(memory_trace_file_dirname, exist_ok=True)
+                    if self.memory_custom_plot:
+                        from composer.profiler.utils import export_memory_timeline_html
+                        export_memory_timeline_html(prof, memory_trace_file_name,
+                                                    torch.cuda.current_device())  # type: ignore
```

Comment:
export_memory_timeline takes `device` as optional[str] but due to  https://github.com/pytorch/pytorch/blob/0692240b90fa7d3c4c43728cbf6a046a64de7725/torch/profiler/profiler.py#L278-L283, we cannot do `str( torch.cuda.current_device()` when we pass in the device value, otherwise the device is not set correctly.